### ISSUE TITLE
apcli0 cann't get ip after wifi reload

### ISSUE THREAD

**Issue by elgent** [Vonger/vocore2#3]
Have anyone meet my problem as below?
After modify the wireless to ap+sta mode correctly, when first start up, it probably work well.
1. exec shell cmd "wifi up";
2. rescan to connect another ssid by luci
3. /etc/init.d/network restart
When i do any of above, the apcli0 can connect success, but can't get ip again.
If i reboot the device now or do the following steps, it probably work well again.
ifconfig apcli0 down
ifconfig ra0 down
rmmod mt7628
insmod mt7628
wifi up

Any suggestion? Thank you.

**Comment by rkkoszewski** [comment]
Same here on a Vocore 2 Lite. Once it connects to an AP and then you restart the wifi, it won't get a new IP unless you restart the device.

**Comment by Vonger** [comment]
currently solution is have to restart...
I have no idea about this bug yet :)

**Comment by rkkoszewski** [comment]
From what I could observe, it doesn't seem to be an issue with the _ip_, but a connection issue.

If you run "iwinfo" while connected it shows the AP and the AP MAC. When you run "wifi" and wait a bit, the AP name will show up but the AP MAC will be 00:00:00...

Also if you set a static IP, you will see that you will get nothing trough.

**Comment by elgent** [comment]
I found not every time 'rmmod + insmod mt7628' take effect now
This problem has puzzled me for a long time :(

**Comment by elgent** [comment]
@rkkoszewski I agree with you
It seem that something is wrong in the driver cause fail to send pkt out...
But i have on idea now after review some driver code

**Comment by rkkoszewski** [comment]
Any update on this? Has there been any work with this? There have been several commits and I'm unable to test it right now.

**Comment by Vonger** [comment]
It should fixed by the new script.

**Comment by rkkoszewski** [comment]
Issue is still present. Completely fresh build following Vocore2.git instructions as per Readme. I go to Luci, connect to an AP (Run ifconfig trough serial and you see an IP), and when I click on "restart" ra0, apcli0 never manages to get an IP again (Run ifconfig trough serial, and no IP in sight). Restart of Vocore2 Lite fixes the issue as always.

**Comment by Vonger** [comment]
I just update new patch ralink.sh has updated.
This time it should be totally fixed :)

**Comment by rkkoszewski** [comment]
I confirm, seems to be fixed now. I just compiled it again from scratch and flashed it on my Vocore 2 Lite. When you reconnect from command line with "wifi" it reconnects and gets the IP. If you use Luci and restart the wifi interface, it resets the wifi and finds an IP also. So it seems like the IP bug after wifi restart is fixed.

One thing I noticed is that when you ONLY disable the wifi client from Luci, the AP data gets copied to the Client and it breaks everything till you restart. But that's a different bug.
